nevergrad/optimization/optimizerlib.py

```diff
@@ -1234,6 +1234,8 @@ def __init__(
         self.max_num_vars = max_num_vars
         self.multivariate_optimizer = multivariate_optimizer
         self.monovariate_optimizer = monovariate_optimizer
+        if self.max_num_vars is not None and self.max_num_vars == 1:  # If we decide to focus on 1, we want 1.
+            self.monovariate_optimizer = multivariate_optimizer
```

Comment:
@teytaud I find this really unexpected, if max_num_var is 1 I would definitely expect the monovariate optimizer to be used!

Reply:
All the mess is because CMA does not like dimension one. Maybe I should just switch from monovariate + multivariate to
multivariate + monovariate_for_cases_in_which_the_multivariate_optimizer_does_not_like_dim_1 and switch to the latter only if the multivariate one does not work in dim 1 ?

Ok the name "monovariate_for_cases_in_which_the_multivariate_optimizer_does_not_like_dim_1" is slightly long.

Reply:
maybe don't make it a parameter and just switch to something else for CMA?

Reply:
I just removed that blocking part and we'll see later.